01/1_1.playground/Contents.swift

```diff
@@ -0,0 +1,79 @@
+/*
+Exercise 1.1.  Below is a sequence of expressions. What is the result printed by the interpreter in response to each expression? Assume that the sequence is to be evaluated in the order in which it is presented.
+*/
+
+
+/*
+10
+*/
+10
+
+/*
+(+ 5 3 4)
+*/
+5 + 3 + 4
+
+/*
+(- 9 1)
+*/
+9 - 1
+
+/*
+(/ 6 2)
+*/
+6 / 2
+
+/*
+(+ (* 2 4) (- 4 6))
+*/
+(2 * 4) + (4 - 6)
+
+/*
+(define a 3)
+*/
+var a = 3
+
+/*
+(define b (+ a 1))
+*/
+var b = a + 1
+
+/*
+(+ a b (* a b))
+*/
+a + b + a + b
+
+/*
+(= a b)
+*/
+a == b
+
+/*
+(if (and (> b a) (< b (* a b)))
+b
+a)
+*/
+b > a && b < (a * b) ? b : a
+
+/*
+(cond ((= a 4) 6)
+((= b 4) (+ 6 7 a))
+(else 25))
+*/
+a == 4 ? 6 : b == 4 ? 6 + 7 + a : 25
+
+/*
+(+ 2
+(if (> b a) b a)
+)
+*/
+2 + (b > a ? b : a)
+
+/*
+(* (cond ((> a b) a)
+((< a b) b)
+(else -1))
+(+ a 1))
+*/
+
+b == a ? -1 : (a > b) ? a : b * (a + 1)
```

Comment:
since `@autoclosure` has to be a parameter, this could be solved using a struct: 

```
struct CondStatement {
    let test:  () -> Bool
    let expression: () -> AnyObject

    init(@autoclosure(escaping) _ test: () -> Bool, @autoclosure(escaping) _ expression: () -> AnyObject) {
        self.test = test
        self.expression = expression
    }
}

func cond(
    statements: [CondStatement],
    @autoclosure fallback: (() -> AnyObject?) = nil) -> AnyObject? {
    for statement in statements {
        if statement.test() {
            return statement.expression()
        }
    }

    return fallback()
}

// then the answer to the last problem is: 

let conditions = [
    CondStatement(a > b, a),
    CondStatement(a < b, b)]

let z = cond(conditions, fallback: -1) as? Int ?? 0 * (a + 1)
```